fdbserver/VersionedBTree.actor.cpp

```diff
@@ -7702,11 +7702,16 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 		        : SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_WINDOW_BYTES;
 
 		EncodingType encodingType = EncodingType::XXHash64;
-
-		// Deterministically enable encryption based on uid
-		if (g_network->isSimulated() && logID.hash() % 2 == 0) {
-			encodingType = EncodingType::XOREncryption;
-			m_keyProvider = std::make_shared<XOREncryptionKeyProvider>(filename);
+		// TODO(yiwu): We should enable encryption only when tenant is in required mode, but currently when existing
+		// storage engine instance is reopen on worker server start, the tenant mode is unknown.
+		if (SERVER_KNOBS->ENABLE_STORAGE_SERVER_ENCRYPTION) {
```

Comment:
Please checkout `EncryptionOpsUtil::isEncryptionSupported`; it is better to consolidate encryption enable logic to one routine, would help us in switching to a db-config encryption control later.
Thoughts?

Reply:
will update. sorry this PR was before the EncryptionOpsUtil change and I forget to update it.

Reply:
`EncryptionOpsUtil::isEncryptionSupported` require ClientDBInfo, which is unavailable when Redwood is being ope by worker server before it register with CC. In this case the correct behavior should be (a) open Redwood using its existing encoding, and (b) When the ClientDBInfo become available, reject the storage engine if ClientDBInfo require an encrypted storage, but the Redwood instance is not. Will update the PR with (a), and do (b) in follow up PR.